src/app/applications/components/applications-list/applications-list.tsx

```diff
@@ -168,6 +168,7 @@ export class ApplicationsList extends React.Component<Props, State> {
                                                         } items={[
                                                             { title: 'Sync', action: () => this.syncApplication(app.metadata.name, 'HEAD') },
                                                             { title: 'Delete', action: () => this.deleteApplication(app.metadata.name, false) },
+                                                            { title: 'Delete (Cascade)', action: () => this.deleteApplication(app.metadata.name, true) },
```

Comment:
Instead of adding secondary delete menu item lets add cascade checkbox to the confirmation message. The PopupApi confirm does not allow adding custom inputs to confirmation form. I'm working on adding this feature.

Reply:
@alexmt awesome!  Is it worth merging the feature and converting it to use the checkbox later, or is this a pretty small deal to add to the PopupApi confirm?

Reply:
It is pretty small. I will finish it today and send you update.

Reply:
@merenbach I've pushed the changes. Usage example is available here https://github.com/argoproj/argo-ui/blob/master/stories/popup.tsx#L45 : 

```
const confirmed = await popupApi.confirm('Do it!', () => (
                            <div>
                                Click checkbox and confirm <Checkbox checked={store.state.checked} onChange={(val) => store.set({ checked: val })} />
                            </div>
                        ));
```